llvm/test/TableGen/MacroFusion.td

```diff
@@ -78,7 +88,7 @@ def TestFusion: SimpleFusion<"test-fusion", "HasTestFusion", "Test Fusion",
 // CHECK-PREDICATOR-NEXT:    }
 // CHECK-PREDICATOR-NEXT:    return true;
 // CHECK-PREDICATOR-NEXT:  }
-// CHECK-PREDICATOR-NEXT:  bool isTestFusion(
```

Comment:
Hi, @wangpc-pp , since shouldScheduleAdjacent may be entered twice, the second time with new created commute instruction which does not have parent MF, so there will be a segment fault to getMF() `auto &MRI = SecondMI.getMF()->getRegInfo();`

Reply:
Can you provide a test case?

Reply:
def TuneMoveConBrFusion  :
  Fusion<"move-fusion", "HasMoveFusion", "Enable MOVE+CONBR macro fusion",
  [SecondFusionPredicateWithMCInstPredicate<CheckOpcode<[BEQ, BGE, BGEU, BLT, BLTU, BNE, C_BEQZ, C_BNEZ]>>,
   WildcardTrue,
   FirstFusionPredicateWithMCInstPredicate<
     CheckAny<[
       CheckOpcode<[C_LI, C_MV, LUI, C_LUI, AUIPC]>,
       CheckAll<[CheckOpcode<[ADDI]>, CheckIsRegOperand<1>, CheckRegOperand<1, X0>]>]>>,
   OneUse,
   TieReg<0, 0>]>;

Reply:
> the second time with new created commute instruction which does not have parent MF

I think the problem is here and it is not related to this PR. I don't know why there is a new created commute instruction without parent since fusion won't create new instruction. Maybe you can provide more informations, like stack trace?

Reply:
Please see scheduleAdjacentImpl in MacroFusion.cpp. It tries to create a NewMI.

Reply:
Which part of `scheduleAdjacentImpl` will create a NewMI?

Reply:
Sorry,  it's downstream code. Never mind!